tests/pyfunc/test_model_export_with_class_and_artifacts.py

```diff
@@ -1533,6 +1463,23 @@ def test_python_model_predict_with_params():
     ]
 
 
+def test_python_model_with_type_hint_errors_with_different_signature():
+    signature = infer_signature(["input1", "input2"], params={"foo": [8]})
+
+    with mock.patch("mlflow.pyfunc._logger.warning") as warn_mock:
+        with mlflow.start_run():
```

Comment:
```suggestion
    with mlflow.start_run():
        with mock.patch("mlflow.pyfunc._logger.warning") as warn_mock:
```

can we swap the mocks?